### PR TITLE
Implement Add/Remove Cmdlets for VMSS diagnostics extension, add rela…

### PR DESCRIPTION
**Pull request by zhencui**
## Comments

---

This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. For the categories that you did not check off, you can remove them from this body. Within each of the categories that you did select, make sure that you can check off **all** of the boxes. 

For information on cleaning up the commits in your pull request, click [here](../documentation/cleaning-up-commits.md).
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)
- [x] [Add/remove/edit cmdlet(s)](#cmdlet-signature)
### General
- [x] Title of the PR is clear and informative
- [x] There are a small number of commits that each have an informative message
- [x] If it applies, references the bug/issue that the PR fixes
- [x] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)
### Tests
- [x] PR includes test coverage for the included changes
- [x] Tests must use xunit, and should either use Moq to mock management client calls, or use the scenario test framework
- [x] PowerShell scripts used in tests must not use hard-coded values for location
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for existing resources
- [x] Tests should not use App.config files for settings
- [x] Tests should use the built-in PowerShell functions for generating random names when unique names are necessary - this will store names in the test recording
- [x] Tests should use Start-Sleep to pause rather than Thread.Sleep
### Cmdlet Signature
- [x] Cmdlet name uses an approved PowerShell verb - use the enums for `VerbsCommon`, `VerbsCommunication`, `VerbsLifecycle`, `VerbsOther` whenever possible
- [x] Cmdlet noun name uses the AzureRm prefix for management cmdlets, and the Azure prefix for data plane cmdlets
- [x] Cmdlet specifies the `OutputType` attribute if any output is produced; if the cmdlet produces no output, it should implement a `PassThrough` parameter
- [x] If the cmdlet makes changes or has side effects, it should implement `ShouldProcess` and have `SupportShouldProcess = true` specified in the cmdlet attribute. See a discussion about correct `ShouldProcess` implementation [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change)
- [x] Cmdlets should derive from AzureRmCmdlet for management cmdlets, and AzureDataCmdlet for data cmdlets
- [x] If multiple parameter sets are implemented, the cmdlet should specify a `DefaultParameterSetName` in its cmdlet attribute
### Parameters
- [x] Cmdlets should have no more than four positional parameters
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet
- [x] Parameters should be explicitly marked as Mandatory or not, and should contain a HelpMessage
- [x] No parameter is of type `object`.
  - Management cmdlets should have the following parameters and aliases:
    - ResourceGroupName with (optional) alias to `ResourceGroup` type string marked as [ValueFromPipelineByPropertyName]
    - Name with alias to `ResourceName` type string marked as [ValueFromPipelineByPropertyName]
    - Location (if appropriate) type string
    - Tag, type `HashTable`
### Parameters and the Pipeline
- [x] Complex parameters should take values from the pipeline when possible, and certainly when they match the output type of another cmdlet
- [x] Only one parameter should use ValueFromPipeline per parameter set; parameters from different parameter sets may have this attribute, but should not be convertible
- [x] No parameter is of type `object`
- [x] Each management cmdlet should have a parameter set that takes `ResourceGroupName` and `Name` from the pipeline by property value
- [x] For a given resource type, it should be possible to pipe the output of `Get` and `New` cmdlets to the input of `Set`, `Update`, `Remove` and other action cmdlets for that resource

…ted test case and scenario records

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@zhencui__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (zhencui). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
@zhencui Hey Zhenhua, please merge the latest changes from dev to fix the conflicts

**Comment by markcowl** [comment]
@zhencui There are merge conflicts, please update

**Comment by zhencui** [comment]
@cormacpayne  @markcowl  Conflicts have been resolved.

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/RemoveAzureRmVmssDiagnosticsExtension.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        public VirtualMachineScaleSet VirtualMachineScaleSet { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true)]
+        public string Name { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/RemoveAzureRmVmssDiagnosticsExtension.cs]: Should the remove have an alias for Name property in order to be consistent with the Add-AzureRmVmssDiagnosticsExtension cmdlet Name property?

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/DiagnosticsExtensionTests.ps1 [hunk 2]
```diff
@@ -183,4 +183,87 @@ function Test-DiagnosticsExtensionSupportJsonConfig
...
+        New-AzureRmStorageAccount -ResourceGroupName $rgname -Name $storagename -Location $loc -Type $storagetype
+
+        $ipCfg = New-AzureRmVmssIPConfig -Name 'test' -SubnetId $subnetId;
+        $vmss = New-AzureRmVmssConfig -Location $loc -SkuCapacity 2 -SkuName 'Standard_A0' -UpgradePolicyMode 'automatic' -NetworkInterfaceConfiguration $netCfg `
+            | Add-AzureRmVmssNetworkInterfaceConfiguration -Name 'test' -Primary $true -IPConfiguration $ipCfg `
+            | Set-AzureRmVmssOSProfile -ComputerNamePrefix 'test' -AdminUsername $adminUsername -AdminPassword $adminPassword `
+            | Set-AzureRmVmssStorageProfile -Name 'test' -OsDiskCreateOption 'FromImage' -OsDiskCaching 'None' `
+            -ImageReferenceOffer $imgRef.Offer -ImageReferenceSku $imgRef.Skus -ImageReferenceVersion $imgRef.Version `
+            -ImageReferencePublisher $imgRef.PublisherName -VhdContainer $vhdContainer `
+            | Add-AzureRmVmssDiagnosticsExtension -Name $extname -SettingFilePath "$TestOutputRoot\ConfigFiles\DiagnosticsExtensionPublicConfig.json" `
```
Comment (shahabhijeet) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/DiagnosticsExtensionTests.ps1]: Any reason, no tests have not been added for covering rest of the parameters for AddAzureRmVmssdianosticsExtension (e.g. VirtualMachineScaleSet, PrtectedSettingFilePath, TypeHandlerVersion)
If you are planning to add those test later, can you create an issue and link to this PR or if you think those tests are not needed, please mention in the PR.

**Comment by cormacpayne** [comment]
@azuresdkci retest this please

**Comment by markcowl** [comment]
@zhencui The PR is failing due to cmdlet signature issues for the two new cmdlets caught by the static analysis tools. I have placed comments in the two cmdlets that need to be changed.  See http://azuresdkci.cloudapp.net/job/powershell/6512/artifact/src/Package/SignatureIssues.csv  for the static analysis report.

**Comment by cormacpayne** [comment]
@zhencui the following test is failing:

`Microsoft.Azure.Commands.Compute.Test.ScenarioTests.DiagnosticsExtensionTests.TestVmssDiagnosticsExtension`

**Comment by cormacpayne** [comment]
@zhencui Hey Zhenhua, would you mind pulling from dev to get the latest changes that fix the on-demand build?

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1203

File: src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/RemoveAzureRmVmssDiagnosticsExtension.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                    // If extension name is provided, should match the name as well
+                    var diagnosticsExtensions = extensions
+                        .Where(extension =>
+                            DiagnosticsHelper.IsDiagnosticsExtension(extension) &&
+                            (string.IsNullOrEmpty(this.Name) || extension.Name.Equals(this.Name, StringComparison.InvariantCultureIgnoreCase)))
+                        .ToList();
+
+                    if (diagnosticsExtensions.Any())
+                    {
+                        if (Force.IsPresent ||
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/RemoveAzureRmVmssDiagnosticsExtension.cs]: Is this really necessary?  Seems liek the ShouldContinue is going to happen every time anythiung is deleted, so not much of a reason for the prompt or for the Force parameter.  I would remove it.

**Comment by markcowl** [comment]
@zhencui One comment on the lack of need for Force parameter.  Otherwise LGTM

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1217/

**Pull request merged**